Created Problem 78 & 90 English Version (#357)
azl397985856/leetcode: problems/78.subsets.en.md

## Problem Link
https://leetcode.com/problems/subsets/description/

## Description
```
Given a set of distinct integers, nums, return all possible subsets (the power set).

Note: The solution set must not contain duplicate subsets.

Example:

Input: nums = [1,2,3]
Output:
[
  [3],
  [1],
  [2],
  [1,2,3],
  [1,3],
  [2,3],
  [1,2],
  []
]

```

## Solution

Since this problem is seeking `Subset` not `Extreme Value`, dynamic programming is not an ideal solution. Other approaches should be taken into our consideration. 

> Actually, there is a general approach to solve problems similar to this one -- backtracking. Given a [Code Template](https://leetcode.com/problems/combination-sum/discuss/16502/A-general-approach-to-backtracking-questions-in-Java-(Subsets-Permutations-Combination-Sum-Palindrome-Partitioning)) here, it demonstrates how backtracking works with varieties of problems. Apart from current one, many problems can be solved by such a general approach. For details, please kindly check Related Problems section below.

Reviewer: ```suggestion
Actually, there is a general approach to solve problems similar to this one -- backtracking. Given a [Code Template](https://leetcode.com/problems/combination-sum/discuss/16502/A-general-approach-to-backtracking-questions-in-Java-(Subsets-Permutations-Combination-Sum-Palindrome-Partitioning)) here, it demonstrates how backtracking works with varieties of problems. Apart from current one, many problems can be solved by such a general approach. For more details, please check the `Related Problems` section below.
```

## Problem Link
https://leetcode.com/problems/subsets/description/

## Description
```
Given a set of distinct integers, nums, return all possible subsets (the power set).

Note: The solution set must not contain duplicate subsets.

Example:

Input: nums = [1,2,3]
Output:
[
  [3],
  [1],
  [2],
  [1,2,3],
  [1,3],
  [2,3],
  [1,2],
  []
]

```

## Solution

> This problem is required to seek`subset`, rather than `Extreme value`. As a result, dynamic programming is not particularly suitable. We need to look into other ways.

Reviewer: ```suggestion
This problem is seeking `subset` rather than `Extreme value`. So dynamic programming is not particularly suitable. We need to look for other solutions.
```

> Actually, there is a general approach to solve problems like this one -- backtracking. An Ace provided a [Code Template](https://leetcode.com/problems/combination-sum/discuss/16502/A-general-approach-to-backtracking-questions-in-Java-(Subsets-Permutations-Combination-Sum-Palindrome-Partitioning)) on backtracking, solved all problems in the link. Apart from current problem, a great number of supposed to be solved by such a general approach. For details, please kindly check Related Problems section below.

Reviewer: ```suggestion
Actually, there is a general approach to solve problems like this one -- backtracking. An Ace provided a [Code Template](https://leetcode.com/problems/combination-sum/discuss/16502/A-general-approach-to-backtracking-questions-in-Java-(Subsets-Permutations-Combination-Sum-Palindrome-Partitioning)) on backtracking, solved all problems in the link. Apart from current problem, a great number of problems can to be solved by such a general approach. For details, please kindly check Related Problems section below.
```

> Let's first look at the problem-solving ideas of the general solution. Giving a picture:

Reviewer: ```suggestion
Firstly，Let's look at the problem-solving ideas of the general solution. Giving a picture:
```
